src/QMCWaveFunctions/tests/test_einset.cpp

```diff
@@ -86,6 +86,104 @@ TEST_CASE("Einspline SPO from HDF diamond_1x1x1", "[wavefunction]")
   // due to the different ordering of bands skip the tests on CUDA+Real builds
   // checking evaluations
   // Reference values can be checked using eval_bspline_spo.py
+    /*
+      Here we test the low-level spline implementation of applyRotation().
+      Manually encode the unitary transformation. Ugly, but it works.
+
+      NB: This is truncated to 5 sig-figs, so there is some slop here as
+      compared to what is done in the splines via apply_rotation().
+      So below we reduce the threshold for comparison. This can
+      probably be ditched once we have a way to grab the actual
+      rotation matrix...
+    */
+    const auto orbitalsetsize = spo->getOrbitalSetSize();
+    REQUIRE(orbitalsetsize == 7);
```

Comment:
your tests are failing here because the "particles" xml input above still has size="8" but you are using REQUIRE(orbitalsetsize == 7); 

Also, you could move all of this stuff into the test_spline_applyrotation.cpp file you added to keep the appyRotation checks contained to one file.